blocks/extensions.py

```diff
@@ -0,0 +1,141 @@
+class TrainingExtension(object):
```

Comment:
What about adding something like "on_timer", that would be fired every number of seconds or examples. GH uses something like that to save every 5 minutes for instance.